lib/portlayer/exec/container.go

```diff
@@ -393,16 +393,24 @@ func infraContainers(ctx context.Context, sess *session.Session) ([]mo.VirtualMa
 
 // find the childVM for this resource pool by name
 func childVM(ctx context.Context, sess *session.Session, name string) (*vm.VirtualMachine, error) {
-	searchIndex := object.NewSearchIndex(sess.Client.Client)
-	child, err := searchIndex.FindChild(ctx, VCHConfig.ResourcePool.Reference(), name)
```

Comment:
Actually, it's not renaming issue on vSAN. We didn't have vmx file path configured on vSAN as I mentioned above, so the file path will be prettyname-ID, so it's not possible to query back that VM with ID only.
I tried the datastore path search, which does not support partial match as well.